File: school/summer2013/example_openmp/fortran/44_offload_n_openmp.f90
program sum_offload_n_openmp3
    use omp_lib
    implicit none
    integer(8) :: i, total_sum = 0, local_sum

!$omp parallel private(local_sum)
    local_sum = 0
    !$omp single
        !dir$ offload begin target(mic)
            print *, 'check'
        !dir$ end offload
    !$omp end single nowait

    !$omp do schedule(dynamic)
    do i=1,N
        local_sum = local_sum + i
    end do
    !$omp end do

    !$omp atomic
    total_sum = total_sum + local_sum
!$omp end parallel
end

File: summer2013/example_openmp/c/44_offload_n_openmp.c
#include <stdio.h>

int main()
{
	long i, sum=0, local_sum;
#pragma omp parallel private(local_sum)
{
	local_sum = 0;
	#pragma omp single nowait
	#pragma offload target(mic)
	{
		printf("check\n");
	}
	#pragma omp for schedule(dynamic)
	for(i=1; i<=N; i++)
		local_sum += i;

	#pragma omp atomic
	sum += local_sum;
}
	printf("sum = %ld\n", sum);
}
